subprojects/architecture-test/src/test/java/org/gradle/architecture/test/ArchUnitFixture.java

```diff
@@ -188,6 +189,16 @@ static DescribedPredicate<JavaMember> annotatedMaybeInSupertypeWith(final Descri
         return new AnnotatedMaybeInSupertypePredicate(predicate);
     }
 
+    static ArchCondition<JavaClass> beAnnotatedMaybeInPackageWith(final Class<? extends Annotation> annotationType) {
+        return ArchConditions.be(annotatedMaybeInPackageWith(annotationType));
+    }
+
+    static DescribedPredicate<JavaClass> annotatedMaybeInPackageWith(final Class<? extends Annotation> annotationType) {
+        String annotationTypeName = annotationType.getName();
```

Comment:
I suppose you can use the class here directly, without going via the name. The `isAnnotatedWith` methods support the a class directly.

Reply:
I wanted to keep `AnnotatedMaybeInPackagePredicate` parametrized with an annotation predicate, rather than just an annotation. But I didn't find a simple way of creating a predicate out of the `Class<? extends Annotation>` instance.
I'd use it, if you can suggest one.

I used the same approach as the neighboring method does:
https://github.com/gradle/gradle/blob/d37f79bc078e816da0417f4b8088299330d88b62/subprojects/architecture-test/src/test/java/org/gradle/architecture/test/ArchUnitFixture.java#L183-L186

Reply:
Alright, fine with me. Do you see a use-case already where we'd use the predicate and not one annotation?

Reply:
Thought about it more, and decided to keep it simple, as you suggested.